pkg/service/one2onerestore/worker.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2025 ScyllaDB
+
+package one2onerestore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/scylladb/go-log"
+	"github.com/scylladb/go-set/strset"
+	"github.com/scylladb/gocqlx/v2"
+	"github.com/scylladb/scylla-manager/v3/pkg/scyllaclient"
+	"github.com/scylladb/scylla-manager/v3/pkg/service/backup/backupspec"
+)
+
+type worker struct {
+	managerSession gocqlx.Session
+
+	client         *scyllaclient.Client
+	clusterSession gocqlx.Session
+
+	logger log.Logger
+}
+
+// getManifestsAndHosts checks that each host in target cluster should have an access to at least one target location and fetches manifest info.
+func (w *worker) getManifestsAndHosts(ctx context.Context, target Target) ([]*backupspec.ManifestInfo, []Host, error) {
+	nodeStatus, err := w.client.Status(ctx)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "nodes status")
+	}
+	var (
+		allManifests  []*backupspec.ManifestInfo
+		nodesCountSet = strset.New()
+	)
+	for _, location := range target.Location {
+		// Ignore location.DC because all mappings should be specified via nodes-mapping file
```

Comment:
That's a good direction, but then we shouldn't allow user to specify location DC in the first place.
Or, we can still treat it as a hint if it simplifies the implementation, although I'm not sure of that.

Reply:
Guys, why not to revert the logic here ?
Instead of iterating over locations, iterate over target nodes to check if they are able to access the location keeping their expected SSTables ? It must have the access.

Actually, please ensure that each node have the access to the corresponding location.

Reply:
It's not the validation stage. My fauly. It's fine.